pyproject.toml

```diff
@@ -16,10 +16,12 @@ keywords = [
 ]
 readme = {file = "README.md", content-type = "text/markdown"}
 license = {text = "MIT"}
-requires-python = ">=3.10"
+requires-python = ">=3.11"
```

Comment:
The datetime format in a STAC item follows the ISO 8601 format `YYYY-MM-DDThh:mm:ssZ` where `Z` stands for UTC. Here is one example (https://earth-search.aws.element84.com/v1/collections/sentinel-2-l2a/items). 

`datetime` can resolve that `Z` since `python >=3.11` (https://www.reddit.com/r/Python/comments/163ugsm/psa_as_of_python_311_datetimefromisoformat/). If we want to stick to lower python versions, I can replace `Z` by `+00:00`. What would you prefer?

Reply:
One possible solution: pandas is already a transitive dependency, so add it as an explicit dependency and use `pd.Timestamp(an_iso8601_string).to_pydatetime()`.

Reply:
@konstntokas , please use pandas and stay with Python 3.10.